src/django/api/models/partner_field.py

```diff
@@ -75,18 +77,62 @@ class Meta:
             'Used when type is "object".'
         )
     )
+    active = models.BooleanField(
+        default=True,
+        help_text=(
+            'Indicates if this partner field is active. '
+            'Inactive fields are not available for contributions and '
+            'will not appear in listings.'
+        )
+    )
+    system_field = models.BooleanField(
+        default=False,
+        help_text=(
+            'Indicates if this is a system field. '
+            'System fields cannot be deleted and have restricted editing.'
+        )
+    )
 
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
 
+    objects = PartnerFieldManager()
+
     def __str__(self):
         return self.name
 
     def save(self, *args, **kwargs):
+        # Check if active status changed or if this is a new field.
+        # Use _state.adding to detect new records.
+        # (works with UUID primary keys).
+        should_invalidate_cache = False
+        is_new = self._state.adding
+
+        if not is_new:
+            try:
+                original = PartnerField.objects \
+                    .get_all_including_inactive() \
+                    .get(pk=self.pk)
+                # Invalidate cache if active status changed.
+                if original.active != self.active:
+                    should_invalidate_cache = True
+            except PartnerField.DoesNotExist:
+                # If original doesn't exist, treat as new.
+                should_invalidate_cache = True
+        else:
+            # New partner field being created.
+            should_invalidate_cache = True
+
         super().save(*args, **kwargs)
-        cache.delete(PARTNER_FIELD_LIST_KEY)
+
+        # Invalidate cache if needed.
+        if should_invalidate_cache:
+            cache.delete(PARTNER_FIELD_LIST_KEY)
+            cache.delete('partner_field_names')
```

Comment:
This logic seems really complicated, can we simplify this?

For example first we can create a function that will invalidate the cache (also maybe create a constant for `partner_field_names` key?):

```python
def invalidate_partner_fields_cache():
  cache.delete(PARTNER_FIELD_LIST_KEY)
  cache.delete('partner_field_names')
```

And then the function can be re written as follows:

```python
if self._state.adding:
    invalidate_partner_field_cache()
    return

try:
    original = PartnerField.objects.get_all_including_inactive().get(
        pk=self.pk
    )

    if original.active != self.active:
        invalidate_partner_field_cache()
except PartnerField.DoesNotExist:
    invalidate_partner_field_cache()
```